### ISSUE TITLE
volume not fully mutable

### ISSUE THREAD

**Issue by benedictstrube** [alberti42/Volume-Control#27]
First of all, great utility! ❤️

One thing I noticed is, that it does not seem to be possible, to fully mute the volume when Volume Control is running. When pressing the mute button or lowering the volume to the minimum, the volume is marginally higher than 0%. Only muting via the vanilla volume slider from the menu bar works.

Maybe this is something you can look into at some point. Not that important tbh.

**Comment by alberti42** [comment]
Thank you for the feedback. Is it possible to send a screenshot? I have never seen this bug before. You can monitor what is happening while you press the volume buttons if you make the app menu visible like in the screenshot.. then you can see whether you effectively reach 0%.

![Bildschirmfoto 2024-08-26 um 11 42 08](https://github.com/user-attachments/assets/ea4cf8ed-14d3-4e5b-943f-2b66781844b7)

**Comment by marckrenn** [comment]
Can confirm, no matter the volume step, the lowest it will ever go for me is {volume 0 + volume step}

**Comment by alberti42** [comment]
It is a bit hard for me to debug because I cannot reproduce the bug. See the uploaded video.

https://github.com/user-attachments/assets/a3848bf0-eba7-46b2-b9fb-5774ca314d2e

**Comment by marckrenn** [comment]
Hey Andrea,

I can confirm that the GUI shows 0% system volume, but if you have media playing, you can still clearly hear it. So the GUI apparently doesn't match the actual volume. This is btw also the case when hitting the hardware mute-key.

I'm on a MBP w/ M1 Pro, if that helps.

Let me know if I can help you debug it somehow. Thanks.

**Comment by marckrenn** [comment]
Just took a brief look at the repo and I think one of these conditionals needs to be <=

https://github.com/alberti42/Volume-Control/blob/b8ea9f3cfa814b6cd98fd3c29410902d0085a00a/Sources/Controllers/ISSoundAdditions.m#L141

https://github.com/alberti42/Volume-Control/blob/b8ea9f3cfa814b6cd98fd3c29410902d0085a00a/Sources/Controllers/ISSoundAdditions.m#L132

**Comment by alberti42** [comment]
Thanks a lot! I will fix and release a new updated version in the next couple of weeks. Since it is hopefully a not supercritical bug, I hope you will have understanding if I do not do it immediately. Thanks again!!

**Comment by marckrenn** [comment]
I have to thank you for keeping your your great tool alive! 🙂

**Comment by alberti42** [comment]
I actually found some time from an airport to fix the bug. You can search for new updates and this should allow you to update the app automatically.

![Bildschirmfoto 2024-10-27 um 18 00 45](https://github.com/user-attachments/assets/4d13b6fa-c549-43ea-a05a-54541f63627e)

Unfortunately I could not test whether the problem is truly fixed because I am not running on a mac right now. If the problem persists, your feedback is welcome.

**Comment by marckrenn** [comment]
This fixed it, thank you so much @alberti42 🎉 

Safe travels and have a nice Sunday! 😊

**Comment by marckrenn** [comment]
Oh no, @alberti42 , it's still there 🙈  It's now barely audible via the MacBook Pro's speaker (a had to place my ear on the speaker in order to hear it), but it's still audible using headphones.

**Comment by alberti42** [comment]
Oh, I am sorry. I was too fast to push out the release then. The problem is that my hears are not good enough to hear anything when I reached minimum volume. I will have to do more debugging. If you have any idea what the problem is, it is much appreciated.

PS: Just to confirm, if you move the volume bar of Apple Music manually and bring it to zero, the sound is gone?

**Comment by marckrenn** [comment]
No worries 😊 With the latest update, it’s noticeably improved—I even thought it was fixed! So it’s not a major issue anymore, though obviously, it would still be great if it were fully resolved.

> If you have any idea what the problem is, it is much appreciated.

I’ll try looking into it, though I don’t have much experience with MacOS internals or Objective-C, so please keep expectations modest.

> PS: Just to confirm, if you move the volume bar of Apple Music manually and bring it to zero, the sound is gone?

Yup, if I do that, it's gone for good.

**Comment by alberti42** [comment]
I have committed a version with debug information about the current and newer volume. You can find the infos in the console of Xcode (if you know how to compile the app) or in the Apple Console app, which you probably can find using spotlight. In the zip file, you find the app already compiled. I am afraid you would have to sign it again.

[Volume Control.zip](https://github.com/user-attachments/files/17610649/Volume.Control.zip)

For some strange reason, Apple forced me to abandon the previous identifier `org.Alberti42.VolumeControlApp`. So I had to compile the app with

`PRODUCT_BUNDLE_IDENTIFIER = com.Alberti42.VolumeControlApp`

I say this because you may wonder why the settings are no longer preserved. When I started this project, it was so simple to compile and deploy apps, because there were not all these security barriers. I am not saying that it is bad thing, but it makes everything so more complicated.

Have a look in the video. You should be able to see that you can reach 0% of the volume in the console:

https://github.com/user-attachments/assets/2b8b5283-54c7-4191-bd09-83a25cac379d

**Comment by alberti42** [comment]
The system volume is now fully mutable in version 2.0.1. If you think the original issue is not solved yet, feel free to reopen the issue.

**Comment by mafredri** [comment]
Awesome, thanks for adding that @alberti42. There's still a minor divergence from OS default where Volume Control doesn't take volume to 0 on that last step, just applies the mute. Functionally there's not much difference, though.

**Comment by alberti42** [comment]
Yes, @mafredri you found it out. I was planning to fix it in the future. In principle, the default behavior offers one last level called 0, which is not muted yet. I will bring this back in the future. It is not really a bug. It works as intended and made a simplification here because there were many more things to update to comply with modern Tahoe.
